packages/opentelemetry-exporter-collector/src/types.ts

```diff
@@ -346,6 +346,7 @@ export interface CollectorExporterConfigBase {
   attributes?: SpanAttributes;
   url?: string;
   concurrencyLimit?: number;
+  beaconBlobPropertyBag?: BlobPropertyBag;
```

Comment:
if not a beaconBlobPropertyBag is defined this will fail... need to provide ability to make it configurable.
this functions will help parse with safety the configured beaconPropertyBag https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-exporter-collector/src/util.ts#L23

Reply:
and this will be the root cause: https://github.com/open-telemetry/opentelemetry-js/blob/9218835d306b0f443f2ac71fcc3b7f191ec2657b/packages/opentelemetry-exporter-collector/src/platform/browser/CollectorExporterBrowserBase.ts#L61, if in the config no value for beaconPropertyBag is defined. 
Hence here you allow an empty implicit object : https://github.com/open-telemetry/opentelemetry-js/blob/9218835d306b0f443f2ac71fcc3b7f191ec2657b/packages/opentelemetry-exporter-collector/src/platform/browser/CollectorExporterBrowserBase.ts#L40

Reply:
This line is the default value
https://github.com/open-telemetry/opentelemetry-js/blob/9218835d306b0f443f2ac71fcc3b7f191ec2657b/packages/opentelemetry-exporter-collector/src/platform/browser/CollectorExporterBrowserBase.ts#L25

And this value is the default value assigned here
https://github.com/open-telemetry/opentelemetry-js/blob/9218835d306b0f443f2ac71fcc3b7f191ec2657b/packages/opentelemetry-exporter-collector/src/platform/browser/CollectorExporterBrowserBase.ts#L59-L62

So, config can be null or undefined, there is a default value.

Reply:
nice , could you extend the test cases when config with beaconBlobPropertyBag is undefined or null ?  that would be nice.  i have written some tests for inspiration that you can see here : https://github.com/open-telemetry/opentelemetry-js/blob/73c873b25ce3e3444ea3658c49129720d5804cd8/packages/opentelemetry-exporter-collector/test/browser/CollectorTraceExporter.test.ts#L250